pkg/config/config.go

```diff
@@ -74,12 +83,97 @@ func ReadConfig(getEnv func(string) string) (*Config, error) {
 	}
 	cfg.MaxHeaderBytes = mhb
 
+	err = loadJWTSigningKeys(getEnv, cfg.JWKeyRegister)
+	if err != nil {
+		return nil, err
+	}
+
 	return &cfg, nil
 }
 
-func loadRESTConfig(kubeconfig string) (*rest.Config, error) {
-	if kubeconfig != "" {
-		return clientcmd.BuildConfigFromFlags("", kubeconfig)
+func loadJWTSigningKeys(getEnv GetEnv, keys jwt.KeyRegister) error {
+	err := loadKeysFromPath(getEnv, keys, "OSB_JWT_KEYS_JWK_URL", loadJWK)
+	if err != nil {
+		return err
+	}
+	err = loadKeysFromPath(getEnv, keys, "OSB_JWT_KEYS_PEM_URL", loadPEM)
+	return err
+}
+
+func loadKeysFromPath(getEnv GetEnv, keys jwt.KeyRegister, envVarName string, loadFunc keyLoadingFun) error {
+	envVarValue := getEnv(envVarName)
+	if envVarValue == "" {
+		return nil
+	}
+
+	content, err := loadContentFromPath(envVarValue)
+	if err != nil {
+		return fmt.Errorf("unable to load keys from '%s' (defined in %s): %w", envVarValue, envVarName, err)
+	}
+
+	_, err = loadFunc(keys, content)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadContentFromPath(path string) ([]byte, error) {
+	if path == "" {
+		return nil, fmt.Errorf("an empty path is not allowed")
+	}
+
+	urlOfPath, err := url.Parse(path)
+	if err != nil {
+		return nil, fmt.Errorf("the value '%s' can't be parsed as url: %w", path, err)
+	}
+
+	var content []byte
+	switch {
+	case urlOfPath.Scheme == "https":
+		content, err = loadContentFromHTTP(urlOfPath)
+	case urlOfPath.Scheme == "file":
+		content, err = loadContentFromFile(urlOfPath)
+	case urlOfPath.Scheme == "http":
+		return nil, fmt.Errorf("the scheme '%s' of '%s' is not supported. Did you mean 'https' instead of 'http'? Supported schemes are 'https' and 'file'", urlOfPath.Scheme, path)
+	default:
+		return nil, fmt.Errorf("the scheme '%s' of '%s' is not supported. Supported schemes are 'https' and 'file'", urlOfPath.Scheme, path)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("content can't be loaded from '%s': %w", urlOfPath, err)
+	}
+
+	return content, nil
+}
+
+func loadContentFromHTTP(url *url.URL) ([]byte, error) {
+	client := http.DefaultClient()
+	urlStr := url.String()
+	response, err := client.Get(urlStr)
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to '%s': '%w", urlStr, err)
+	}
+
+	content, err := io.ReadAll(response.Body)
```

Comment:
According to the documentation of [`Response.Body`](https://golang.org/pkg/net/http/#Response), the caller is responsible to close the body, so you probably want something like

```suggestion
	defer response.Body.Close()
	content, err := io.ReadAll(response.Body)
```

Since reading the keys from an URL is only done at most once when the broker is starting up, the original code does not present a particularly problematic leak, but it's still good to adhere to the documented responsibilities we have as the caller.